File: swift/ITSClient/Sources/ITSMobility/Models/Origin.swift
/*
 * Software Name : ITSClient
 * SPDX-FileCopyrightText: Copyright (c) Orange SA
 * SPDX-License-Identifier: MIT
 *
 * This software is distributed under the MIT license,
 * see the "LICENSE.txt" file for more details or https://opensource.org/license/MIT/
 *
 * Software description: Swift ITS client.
 */

import Foundation

/// The entity responsible for this message.
public enum Origin: String, Codable {
    case globalApplication = "global_application"
    case mecApplication = "mec_application"
    case onBoardApplication = "on_board_application"
    case originSelf = "self"
}

File: java/iot3/mobility/src/main/java/com/orange/iot3mobility/message/cam/v113/model/Origin.java
package com.orange.iot3mobility.message.cam.v113.model;

/**
 * Origin v1.1.3
 * <p>
 * The entity responsible for this message.
 */
public enum Origin {
    SELF("self"),
    GLOBAL_APPLICATION("global_application"),
    MEC_APPLICATION("mec_application"),
    ON_BOARD_APPLICATION("on_board_application");

    public final String value;

    Origin(String value) {
        this.value = value;
    }
}
